src/main/java/com/github/danielflower/mavenplugins/release/Reactor.java

```diff
@@ -190,20 +191,25 @@ static AnnotatedTag hasChangedSinceLastRelease(List<AnnotatedTag> previousTagsFo
         try {
             if (previousTagsForThisModule.size() == 0) return null;
             boolean hasChanged = detector.hasChangedSince(relativePathToModule, project.getModel().getModules(), previousTagsForThisModule);
-            return hasChanged ? null : tagWithHighestBuildNumber(previousTagsForThisModule);
+            return hasChanged ? null : findTagWithHighestBuildNumber(previousTagsForThisModule);
         } catch (Exception e) {
             throw new MojoExecutionException("Error while detecting whether or not " + project.getArtifactId() + " has changed since the last release", e);
         }
     }
 
-    private static AnnotatedTag tagWithHighestBuildNumber(List<AnnotatedTag> tags) {
-        AnnotatedTag cur = null;
+    // Ignores any tags with non-numeric build numbers and finds the one with the largest build number among the others.
+    // The tags are guaranteed to be for the same /business version/ i.e. it's likely a small list.
+    private static AnnotatedTag findTagWithHighestBuildNumber(List<AnnotatedTag> tags) {
+        AnnotatedTag tagWithHighestBuildNumber = null;
         for (AnnotatedTag tag : tags) {
-            if (cur == null || tag.buildNumber() > cur.buildNumber()) {
-                cur = tag;
+            if(StringUtils.isNumeric(tag.buildNumber())) {
```

Comment:
As per comment in pom, I'd rather avoid dependencies if possible. The javadoc for this also says that this returning true doesn't guarantee Long.parseLong will work (e.g. if it's a really big number). It's fine to just not check and just try/catch the parseLong. There are no performance concerns etc here.

Reply:
Removed the _use_ of the dependency.